app/src/main/java/fr/neamar/kiss/forwarder/ExperienceTweaks.java

```diff
@@ -91,18 +95,62 @@ public boolean onSingleTapUp(MotionEvent e) {
             public boolean onFling(MotionEvent e1, MotionEvent e2, float velocityX, float velocityY) {
                 float directionY = e2.getY() - e1.getY();
                 float directionX = e2.getX() - e1.getX();
-                if (Math.abs(directionX) > Math.abs(directionY)) {
-                    return false;
-                }
                 if (!isGesturesEnabled()) {
                     return false;
                 }
-                if (directionY > 0) {
-                    // Fling down: display notifications
-                    displayNotificationDrawer();
+
+                if (Math.abs(directionX) > Math.abs(directionY)) {
+                    if (directionX > 0) {
+                        // Swipe right
+                        if (GESTURE_ACTION_SWIPE_RIGHT == 1)
```

Comment:
might be worth creating a function that takes a `gesture` as a parameter and apply the action? Since this piece of code is repeated 4 times